### ISSUE TITLE
Mutation triggered twice for single HTTP request

### ISSUE THREAD

**Issue by Queatz** [apollographql/apollo-server#2589]
I have a mutation that is being called twice for a single HTTP request.  This only happens when the mutation takes longer than 60 seconds.  What's going on here?

**Comment by trevor-scheer** [comment]
Hey @Queatz, thanks for submitting an issue! Could you please provide a runnable reproduction that demonstrates the problem? Thanks!

**Comment by yashspr** [comment]
I have the same issue. The mutation runs for a long time in my case. The mutation resolver was getting called twice because a 2nd POST request was being made to the server. But the client was making only a single request which was evident from the Network tab in browser developer tools.

As I found out, the issue was not caused by apollo server or the client. 

After a lot of research, I found that **Node.js server by default times out a request after 120 seconds** and closes the connection to the client. This in turn causes the **[browser to retry the request](https://stackoverflow.com/a/19697278/6385825)** but the browser doesn't log that retried request in the network tab in developer tools which is a cause of lot of confusion.

So, **[changing the request timeout duration](https://stackoverflow.com/a/58372968/6385825)** in the ExpressJS server resolved the issue for me.

@Queatz I think this was probably what happened in your case as well.
